src/Sandbox.php

```diff
@@ -350,6 +355,21 @@ public function getUrl(): ?string
         return $this->url;
     }
 
+    public function setBranchId(?string $branchId): self
+    {
+        if ($branchId === '') {
+            $branchId = null;
+        }
+
+        $this->branchId = $branchId;
```

Comment:
`$this->branchId = !empty($branchId) ? $branchId : null` ?

Reply:
That would convert `'0'` to `null` because (`empty('0') === true`). Not sure it a branch can has such id, but technically it's a valid branchId.

Reply:
right!  good point, llook at me trying to insert bugs during the review 😂